File: wear/src/main/java/ca/joshstagg/slate/SlateWatchFaceWearablePrefActivity.java
package ca.joshstagg.slate;

import android.os.Bundle;

import preference.WearPreferenceActivity;

/**
 * Slate ca.joshstagg.slate
 * Copyright 2017  Josh Stagg
 */
public class SlateWatchFaceWearablePrefActivity extends WearPreferenceActivity {

    private ConfigService mSlateConfigService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.preferences);
        mSlateConfigService = Slate.getInstance().getConfigService();
        mSlateConfigService.connect();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mSlateConfigService.disconnect();
    }
}

File: wear/src/main/java/ca/joshstagg/slate/SlateWatchFaceWearablePrefActivity.kt
package ca.joshstagg.slate

import android.os.Bundle

import preference.WearPreferenceActivity

/**
 * Slate ca.joshstagg.slate
 * Copyright 2017  Josh Stagg
 */
class SlateWatchFaceWearablePrefActivity : WearPreferenceActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.preferences)
        Slate.instance?.configService?.connect()
    }

    override fun onDestroy() {
        super.onDestroy()
        Slate.instance?.configService?.disconnect()
    }
}
